hm_pyhelper/diagnostics/diagnostics_report.py

```diff
@@ -33,10 +33,33 @@ def __init__(self, diagnostics=[], **kwargs):
 
         if DIAGNOSTICS_ERRORS_KEY not in self:
             self.__setitem__(DIAGNOSTICS_ERRORS_KEY, [])
+
         self.diagnostics = diagnostics
 
+        # Needed for more verbose messages at HPT end. This attribute gets
+        # queried to display exactly which keys failed causing the diagnostics
+        # to fail.
+        self.manufacturing_errors = []
+
     def passed(self):
-        return self[DIAGNOSTICS_PASSED_KEY]
+        # Check only a subset of keys that are required for manufacturing
+        # tests. If these don't have errors then we can conclude that device
+        # is good from manufacturing point-of-view.
+        keys_to_check = {
+            'ECC', 'OK', 'E0', 'W0', 'PK', 'BT', 'VA', 'FR', 'serial_number'
+        }
+
+        errors = self.get_errors()
+        if errors:
+            self.set_passed(False)
```

Comment:
No instance variables should be set while processing this method. The results will be dependent on the `keys_to_check` param that client provides, so it will be context specific and cannot be associated with the instance overall.

Reply:
Sorry I couldn't follow. Can you explain the above paragraph a bit more? If this method is not going to change anything (and does not take any parameters) then perhaps is better suited to be a `@property`?

Overall I need more context on where DiagnosticsReport is being used, if it's only in HPT then changes can be made more freely. If we end up updating hm-diag too then it's a much bigger change.